### PR TITLE
Revise doc comment fix

### PR DESCRIPTION
**Pull request by munificent**

### CODE REVIEW

**Comment by munificent** [comment]
@a14n has committed stuff plenty of times and is A-OK on the CLA.

File: lib/src/chunk_builder.dart [hunk 1]
```diff
@@ -333,6 +341,48 @@ class ChunkBuilder {
     preserveNewlines(linesBeforeToken);
   }
 
+  /// Writes the text of [comment].
+  ///
+  /// If it's a JavaDoc comment that should be fixed to use `///`, fixes it.
+  void _writeCommentText(SourceComment comment) {
+    if (_formatter.fixes.contains(StyleFix.docComments)) {
+      var match = _javaDocComment.firstMatch(comment.text);
+      if (match != null) {
```
Comment (natebosch) [inline lib/src/chunk_builder.dart]: [optional] invert the condition and short circuit return would reduce nesting in the rest of the method.

Same with the condition on line 348.

File: lib/src/chunk_builder.dart [hunk 2]
```diff
@@ -333,6 +341,48 @@ class ChunkBuilder {
...
+        if (lines.isNotEmpty && lines.last.trim().isEmpty) lines.removeLast();
+
+        // Remove a leading "*" from the middle lines.
+        for (var i = 0; i < lines.length; i++) {
+          var line = lines[i];
+          var match = _javaDocLine.firstMatch(line);
+          if (match != null) {
+            line = match.group(1);
+          } else {
+            line = line.trimLeft();
```
Comment (natebosch) [inline lib/src/chunk_builder.dart]: we're potentially changing indentation for lines that don't have a leading `*`. I think that's probably fine if that was the intention, but we can maybe call this out in a comment somewhere.

Comment (munificent) [reply lib/src/chunk_builder.dart]: Done.

**Pull request merged**